cli/preset/cloud_preset.go

```diff
@@ -188,57 +220,139 @@ connectInject:
-		secretKeyGossipKey, secretNameBootstrapToken, secretKeyBootstrapToken,
-		secretNameHCPConfig, cfg.BootstrapResponse.Cluster.BootstrapExpect, secretNameServerCert)
+`, cfg.BootstrapResponse.Cluster.ID, secretNameServerCA, corev1.TLSCertKey,
+		secretNameGossipKey, secretKeyGossipKey, secretNameBootstrapToken,
+		secretKeyBootstrapToken,
+		secretNameHCPResourceID, secretKeyHCPResourceID,
+		secretNameHCPClientID, secretKeyHCPClientID,
+		secretNameHCPClientSecret, secretKeyHCPClientSecret,
+		apiHostCfg, authURLCfg, scadaAddressCfg,
+		cfg.BootstrapResponse.Cluster.BootstrapExpect, secretNameServerCert)
 	valuesMap := config.ConvertToMap(values)
 	return valuesMap
 }
 
 // saveSecretsFromBootstrapConfig takes the following items from the
 // agent bootstrap config from HCP and saves them into known secret names and
 // keys:
-// - HCP config (resource-id, client-id, client-secret).
+// - HCP configresource-id.
+// - HCP client-id.
+// - HCP client-secret.
+// - HCP auth URL (optional)
+// - HCP api hostname (optional)
+// - HCP scada address (optional)
 // - ACL bootstrap token.
 // - gossip encryption key.
 // - server tls cert and key.
 // - server CA cert.
 func (i *CloudPreset) saveSecretsFromBootstrapConfig(config *CloudBootstrapConfig) error {
+	// create namespace
 	if err := i.createNamespaceIfNotExists(); err != nil {
 		return err
 	}
 
-	i.UI.Output(fmt.Sprintf("Saving HCP configuration as secrets in %s namespace", i.KubernetesNamespace), terminal.WithHeaderStyle())
-	if err := i.saveServerHCPConfigSecret(config); err != nil {
+	// HCP resource id
+	data := map[string][]byte{
+		secretKeyHCPResourceID: []byte(config.HCPConfig.ResourceID),
+	}
+	if err := i.saveSecret(secretNameHCPResourceID, data, corev1.SecretTypeOpaque); err != nil {
+		return err
+	}
+	i.UI.Output(fmt.Sprintf("HCP resource id saved in '%s' secret in namespace '%s'.",
+		secretKeyHCPResourceID, i.KubernetesNamespace), terminal.WithSuccessStyle())
+
+	// HCP client id
+	data = map[string][]byte{
+		secretKeyHCPClientID: []byte(config.HCPConfig.ClientID),
+	}
+	if err := i.saveSecret(secretNameHCPClientID, data, corev1.SecretTypeOpaque); err != nil {
+		return err
+	}
+	i.UI.Output(fmt.Sprintf("HCP client id saved in '%s' secret in namespace '%s'.",
+		secretKeyHCPClientID, i.KubernetesNamespace), terminal.WithSuccessStyle())
+
+	// HCP client secret
+	data = map[string][]byte{
+		secretKeyHCPClientSecret: []byte(config.HCPConfig.ClientSecret),
+	}
+	if err := i.saveSecret(secretNameHCPClientSecret, data, corev1.SecretTypeOpaque); err != nil {
 		return err
 	}
-	i.UI.Output(fmt.Sprintf("HCP config saved in '%s' secret in namespace '%s'.",
-		secretNameHCPConfig, i.KubernetesNamespace), terminal.WithSuccessStyle())
+	i.UI.Output(fmt.Sprintf("HCP client secret saved in '%s' secret in namespace '%s'.",
+		secretKeyHCPClientSecret, i.KubernetesNamespace), terminal.WithSuccessStyle())
 
-	if err := i.saveBootstrapTokenSecret(config); err != nil {
```

Comment:
got rid of the small helper functions like `saveBootstrapTokenSecret()` because they really did not do much and I moved everything in line so its quicker to read and know that they are all behaving the same way.